src/app/users/users.component.ts

```diff
@@ -1,4 +1,45 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, Directive, EventEmitter, Input, OnInit, Output, QueryList, ViewChildren } from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+import { Observable, of } from 'rxjs';
+import { map, startWith } from 'rxjs/operators';
+import { UsersService } from '../services/users.service';
+import { IUserItem } from './user-item-model';
+import { MdbModalRef, MdbModalService } from 'mdb-angular-ui-kit/modal';
+import { CreateUserComponent } from './create-user/create-user.component';
+
+
+
+export type SortColumn = keyof IUserItem | '';
+export type SortDirection = 'asc' | 'desc' | '';
+const rotate: { [key: string]: SortDirection } = { 'asc': 'desc', 'desc': '', '': 'asc' };
+
+const compare = (v1: any, v2: any) => v1 < v2 ? -1 : v1 > v2 ? 1 : 0;
```

Comment:
I think it should be moved to separate directive file just to keep component clean